src/PactNet.AspNetCore/ProviderState/ProviderStateMiddleware.cs

```diff
@@ -0,0 +1,133 @@
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+using PactNet.Verifier.ProviderState;
+
+namespace PactNet.AspNetCore.ProviderState
+{
+    /// <summary>
+    /// Defines the provider state middleware
+    /// It provides an endpoint to trigger the provider states before verification of pact interaction
+    /// </summary>
+    public class ProviderStateMiddleware
+    {
+        /// <summary>
+        /// The next pointer
+        /// </summary>
+        private readonly RequestDelegate next;
+        
+        /// <summary>
+        /// The provider state middleware options
+        /// </summary>
+        private readonly ProviderStateOptions options;
+
+        /// <summary>
+        /// The provider state accessor
+        /// </summary>
+        private readonly IProviderStateAccessor providerStateAccessor;
+
+        ///// <summary>
+        ///// the after interaction callback
+        ///// </summary>
+        //private Dictionary<string, Action> afterInteractionCallbacks;
+
+        /// <summary>
+        /// Creates an instance of <see cref="ProviderStateMiddleware"/>
+        /// </summary>
+        /// <param name="options">the middleware options</param>
+        /// <param name="providerStateAccessor">the provider state accessor</param>
+        /// <param name="next">the next handle</param>
+        public ProviderStateMiddleware(IOptions<ProviderStateOptions> options, IProviderStateAccessor providerStateAccessor, RequestDelegate next)
+        {
+            this.options = options.Value;
+            this.providerStateAccessor = providerStateAccessor;
+            this.next = next;
+        }
+
+        public async Task InvokeAsync(HttpContext context)
+        {
+            if (!(context.Request.Path.Value?.StartsWith(options.RouteProviderState) ?? false))
+            {
+                await this.next(context);
+                return;
+            }
+
+            if (context.Request.Method != HttpMethod.Post.ToString())
+            {
+                await WriteErrorInResponseAsync(context, "The provider state invocation needs to be a post method. Check on your verifier side if the provider state url is nicely configured.", HttpStatusCode.BadRequest);
```

Comment:
This should have status code 405 - Method Not Allowed: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/405

Reply:
Noted